src/providers/WorkflowCore.Providers.Azure/ServiceCollectionExtensions.cs

```diff
@@ -28,8 +28,9 @@ public static WorkflowOptions UseAzureSynchronization(this WorkflowOptions optio
 
         public static WorkflowOptions UseCosmosDbPersistence(this WorkflowOptions options, string connectionString, string databaseId)
         {
-            options.Services.AddTransient<ICosmosDbProvisioner>(sp => new CosmosDbProvisioner(connectionString, sp.GetService<ILoggerFactory>()));
-            options.UsePersistence(sp => new CosmosDbPersistenceProvider(connectionString, databaseId, sp.GetService<ICosmosDbProvisioner>()));
+            options.Services.AddSingleton<ICosmosDbClient>(sp => new CosmosDbClient(connectionString));
+            options.Services.AddTransient<ICosmosDbProvisioner>(sp => new CosmosDbProvisioner(sp.GetService<ICosmosDbClient>(), sp.GetService<ILoggerFactory>()));
```

Comment:
Do we still need the lambda here?

Reply:
I think yes. Let it disposal automatically.
See
https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection#service-registration-methods

Reply:
I mean
```c#
 options.Services.AddTransient<ICosmosDbProvisioner>();
```